src/gepa/examples/mcp_tool_optimization/cloud_api.py

```diff
@@ -0,0 +1,209 @@
+            "reference_answer": "sample project",
+            "additional_context": {},
+        },
+        {
+            "user_query": "What data is in the CSV file?",
+            "tool_arguments": {"path": f"{temp_dir}/data.csv"},
+            "reference_answer": "Alice",
+            "additional_context": {},
+        },
+        {
+            "user_query": "Tell me about the debug setting in config",
+            "tool_arguments": {"path": f"{temp_dir}/config.json"},
+            "reference_answer": "debug",
+            "additional_context": {},
+        },
+    ]
+    return dataset
+
+
+def simple_metric(item, output: str) -> float:
+    """
+    Simple metric: check if reference answer appears in output.
+
+    Returns 1.0 if reference is found, 0.0 otherwise.
+    """
+    if item["reference_answer"] and item["reference_answer"].lower() in output.lower():
+        return 1.0
+    return 0.0
+
+
+def main():
+    """Run MCP tool optimization example."""
+    print("=" * 60)
+    print("GEPA MCP Tool Optimization Example")
+    print("=" * 60)
+
+    # Create test environment
+    temp_dir = create_test_files()
+    dataset = create_dataset(temp_dir)
+
+    print(f"\nDataset size: {len(dataset)} examples")
+    print(f"Example query: {dataset[0]['user_query']}")
+
+    # Configure MCP server (filesystem server)
+    server_params = StdioServerParameters(
+        command="npx",
+        args=["-y", "@modelcontextprotocol/server-filesystem", temp_dir],
+    )
+
+    print("\nMCP Server Configuration:")
+    print(f"  Command: {server_params.command}")
+    print(f"  Args: {' '.join(server_params.args)}")
+
+    # Create adapter
+    adapter = MCPAdapter(
+        server_params=server_params,
+        tool_name="read_file",
+        task_model="openai/gpt-4o-mini",
+        metric_fn=simple_metric,
+        base_system_prompt="You are a helpful file assistant.",
+        enable_two_pass=True,
+    )
+
+    print("\nAdapter Configuration:")
+    print("  Tool: read_file")
+    print("  Model: openai/gpt-4o-mini")
+    print("  Two-pass: Enabled")
+
+    # Seed candidate with basic tool description
+    seed_candidate = {
+        "tool_description": "Read the contents of a file from the filesystem.",
+    }
+
+    print("\nSeed Candidate:")
+    print(f"  Tool Description: {seed_candidate['tool_description']}")
+
+    # Split dataset
+    trainset = dataset[:3]
+    valset = dataset[3:]
+
+    print("\nDataset Split:")
+    print(f"  Training: {len(trainset)} examples")
+    print(f"  Validation: {len(valset)} examples")
+
+    # Run optimization
+    print("\n" + "=" * 60)
+    print("Starting Optimization...")
+    print("=" * 60 + "\n")
+
+    try:
+        result = gepa.optimize(
+            seed_candidate=seed_candidate,
+            trainset=trainset,
+            valset=valset,
+            adapter=adapter,
+            reflection_lm="openai/gpt-4o",  # Just an example replace your model
```

Comment:
GEPA examples should be using the latest models ideally. Can we replace gpt-4o with at least 4.1 or even 5?